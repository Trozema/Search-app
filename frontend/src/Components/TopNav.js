import React from 'react';

//This is the navigation bar which uses react bootstrap//
import {Navbar, Nav, NavDropdown,} from 'react-bootstrap';

function TopNav(){
    return(
<Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
  <Navbar.Brand href="/">Itunes Search App</Navbar.Brand>
  <Navbar.Toggle aria-controls="responsive-navbar-nav" />
  <Navbar.Collapse id="responsive-navbar-nav">
    <Nav className="mr-auto">
      <NavDropdown title="Favourites" id="collasible-nav-dropdown">
      <NavDropdown.Item href="/favourites">View Favourites</NavDropdown.Item>  
      </NavDropdown>
    </Nav>
  </Navbar.Collapse>
</Navbar>
    )
}


   


export default TopNav;
