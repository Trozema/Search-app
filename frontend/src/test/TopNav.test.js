import React from 'react';
import renderer from 'react-test-renderer';
import TopNav from '../Components/TopNav';

it ("renders successfully", () => {
   const tree = renderer.create(<TopNav/>).toJSON();
   expect(tree).toMatchSnapshot()

})
