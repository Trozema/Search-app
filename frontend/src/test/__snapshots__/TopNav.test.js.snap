// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Itunes Search App
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          data-rb-event-key={null}
          href="#"
          id="collasible-nav-dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Favourites
        </a>
      </div>
    </div>
  </div>
</nav>
`;
